beets/util/hidden.py

```diff
@@ -18,69 +19,41 @@
 import os
 import stat
 import sys
+from pathlib import Path
+from typing import Union
 
-import beets.util
 
-
-def _is_hidden_osx(path):
-    """Return whether or not a file is hidden on OS X.
-
-    This uses os.lstat to work out if a file has the "hidden" flag.
+def is_hidden(path: Union[bytes, Path]) -> bool:
     """
-    file_stat = os.lstat(beets.util.syspath(path))
-
-    if hasattr(file_stat, "st_flags") and hasattr(stat, "UF_HIDDEN"):
-        return bool(file_stat.st_flags & stat.UF_HIDDEN)
-    else:
-        return False
-
-
-def _is_hidden_win(path):
-    """Return whether or not a file is hidden on Windows.
-
-    This uses GetFileAttributes to work out if a file has the "hidden" flag
-    (FILE_ATTRIBUTE_HIDDEN).
+    Determine whether the given path is treated as a 'hidden file' by the OS.
     """
-    # FILE_ATTRIBUTE_HIDDEN = 2 (0x2) from GetFileAttributes documentation.
-    hidden_mask = 2
 
-    # Retrieve the attributes for the file.
-    attrs = ctypes.windll.kernel32.GetFileAttributesW(beets.util.syspath(path))
+    if isinstance(path, bytes):
+        path = Path(os.fsdecode(path))
 
-    # Ensure we have valid attributes and compare them against the mask.
-    return attrs >= 0 and attrs & hidden_mask
+    # TODO: Avoid doing a platform check on every invocation of the function.
+    # TODO: Stop supporting 'bytes' inputs once 'pathlib' is fully integrated.
 
+    if sys.platform == "win32":
+        # On Windows, we check for an FS-provided attribute.
 
-def _is_hidden_dot(path):
-    """Return whether or not a file starts with a dot.
+        # FILE_ATTRIBUTE_HIDDEN = 2 (0x2) from GetFileAttributes documentation.
+        hidden_mask = 2
 
-    Files starting with a dot are seen as "hidden" files on Unix-based OSes.
-    """
-    return os.path.basename(path).startswith(b".")
-
-
-def is_hidden(path):
-    """Return whether or not a file is hidden. `path` should be a
-    bytestring filename.
+        # Retrieve the attributes for the file.
+        attrs = ctypes.windll.kernel32.GetFileAttributesW(str(path))
 
-    This method works differently depending on the platform it is called on.
+        # Ensure we have valid attributes and compare them against the mask.
+        return attrs >= 0 and attrs & hidden_mask
```

Comment:
Perhaps add a variable or something as an intermediate step here with `attrs` and `hidden_mask`. Mixing logical AND with `and` and the bitwise AND with `&` seems like a recipe for misunderstanding down the road when someone next looks at this part of the code.

Reply:
Ooh yeah, I can do that!